R/backend-dbplyr__duckdb_connection.R

```diff
@@ -78,13 +78,22 @@ duckdb_n_distinct <- function(..., na.rm = FALSE) {
   check_dots_unnamed()
 
   if (!identical(na.rm, FALSE)) {
-    stop("Parameter `na.rm = TRUE` in n_distinct() is currently not supported in DuckDB backend.", call. = FALSE)
-  }
+    cols <- list(...)
+
+    # check for more than one vector argument: only one vector is supported
+    # Why not use ROW() as well? Because duckdb's FILTER clause does not support
+    # a windowing context as of now: https://duckdb.org/docs/sql/query_syntax/filter.html
+    if (length(cols) > 1) {
+      stop("n_distinct(): Only one vector argument is currently supported when `na.rm = TRUE`.", call. = FALSE)
```

Comment:
I wouldn't understand "vector argument" when seeing this error, but it makes sense in the context of this PR and it's also easy to find.